src/IonText.ts

```diff
@@ -267,51 +267,44 @@ function toCharCodes(s: string) {
   return charCodes;
 }
 
-function unicodeEscape(codePoint: number) : number[] {
-  let prefix: number[] = [CharCodes.BACKSLASH, CharCodes.LOWERCASE_U];
-  let hexEscape: string = codePoint.toString(16);
-  while (hexEscape.length < 4) {
-    hexEscape = "0" + hexEscape;
-  }
-  return prefix.concat(toCharCodes(hexEscape));
+function hexEscape(codePoint: number) : number[] {
+    let prefix: number[] = [CharCodes.BACKSLASH, CharCodes.LOWERCASE_X];
```

Comment:
Consider promoting `prefix` to a static value to avoid the cost of allocation/initialization on each call. The call to [`concat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) below will create a new `Array` instance anyway.

Reply:
Promoted to `_HEX_ESCAPE_PREFIX`